devicetypes/smartthings/zigbee-ceiling-fan-light.src/led-fan-lightings.groovy

```diff
@@ -0,0 +1,167 @@
+			}
+			def fanSpeedEvent = createEvent(name: "fanSpeed", value: zigbeeMap.value as Integer)
+			childDevice.sendEvent(fanSpeedEvent)
+			if (fanSpeedEvent.value == 0) {
+				childDevice.sendEvent(name: "switch", value: "off")
+				childDevice.sendEvent(name: "level", value: 0)	// For cloud to cloud device UI update
+			} else {
+				childDevice.sendEvent(name: "switch", value: "on")
+				def int_v = fanSpeedEvent.value
+				int_v = int_v * 25
+				int_v = int_v > 100 ? 100 : int_v
+				childDevice.sendEvent(name: "level", value: int_v)	// For cloud to cloud device UI update
+			}
+		}		
+	} else {
+		def cluster = zigbee.parse(description)
+		if (cluster && 
+		    cluster.clusterInt == ON_OFF_CLUSTER_VALUE &&
+		    cluster.command == 0x07) {
+			if (cluster.data[0] == 0x00) {
+				sendEvent(name: "checkInterval", value: 60 * 12, displayed: false, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID])
+			}
+		}
+	}
+}
+
+def off() {
+	zigbee.off()
+}
+
+def on() {
+	zigbee.on()
+}
+
+def setLevel(value, duration) {
+	zigbee.setLevel(value)
+}
+
+def sendFanSpeed(val) {
+	delayBetween([zigbee.writeAttribute(FAN_CLUSTER_VALUE, FAN_STATUS_VALUE, DataType.ENUM8, val), zigbee.readAttribute(FAN_CLUSTER_VALUE, FAN_STATUS_VALUE)], 100)
+}
+
+def ping() {
+	// PING is used by Device-Watch in attempt to reach the Device	
+	return zigbee.onOffRefresh() +
+		zigbee.readAttribute(FAN_CLUSTER_VALUE, FAN_STATUS_VALUE)
+}
+
+def refresh() {
+	zigbee.onOffRefresh() +
+	zigbee.levelRefresh() +	
+	zigbee.readAttribute(FAN_CLUSTER_VALUE, FAN_STATUS_VALUE)
+}
+
+def configure() {
+	// Device-Watch allows 2 check-in misses from device + ping (plus 1 min lag time)
+	// enrolls with default periodic reporting until newer 5 min interval is confirmed
+	sendEvent(name: "checkInterval", value: 2 * 10 * 60 + 1 * 60, displayed: false, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID])
+	// OnOff minReportTime 0 seconds, maxReportTime 5 min. Reporting interval if no activity
+	return zigbee.onOffConfig(0, 300) +
+		zigbee.levelConfig() +
+		refresh()
+}
+
+def installed() {
+	addChildFan()
+}
+
+def addChildFan() {
+	def componentLabel
+	if (device.displayName.endsWith(' Light') ||
+	    device.displayName.endsWith(' light')) {
+		componentLabel = "${device.displayName[0..-6]} Fan"
+	} else {
+		// no '1' at the end of deviceJoinName - use 2 to indicate second switch anyway
+		componentLabel = "$device.displayName Fan"
+	}	
+	try {
+		String dni = "${device.deviceNetworkId}:1"
+		addChildDevice("ITM Fan Child", dni, device.hub.id, [completedSetup: true, label: "${componentLabel}", isComponent: false])
+	} catch(e) {
+		log.warn "Failed to add ITM Fan Controller - $e"
+	}
+    	def childDevice = getChildDevices()?.find {
+		//find light child device
+        	it.device.deviceNetworkId == "${device.deviceNetworkId}:1" 
+    	}
+	if (childDevice != null) {
+		childDevice.sendEvent(name: "switch", value: "off")
+	}
+}
```

Comment:
Change this to:
```
def addChildFan() {
	def componentLabel
	def childDevice

	if (device.displayName.endsWith(' Light') ||
	    device.displayName.endsWith(' light')) {
		componentLabel = "${device.displayName[0..-6]} Fan"
	} else {
		// no '1' at the end of deviceJoinName - use 2 to indicate second switch anyway
		componentLabel = "$device.displayName Fan"
	}	

	try {
		String dni = "${device.deviceNetworkId}:1"
		childDevice = addChildDevice("ITM Fan Child", dni, device.hub.id, [completedSetup: true, label: "${componentLabel}", isComponent: false])
	} catch(e) {
		log.warn "Failed to add ITM Fan Controller - $e"
	}

	if (childDevice != null) {
		childDevice.sendEvent(name: "switch", value: "off")
	}
}
```

Reply:
Ah, OK! I change it. Thank you so much :)